### PR TITLE
Add Native iOS Support

### PR DESCRIPTION
**Pull request by mattjohnsonpint**
Add platform support for the `net6.0-ios` target by bundling the [Sentry Cocoa SDK](https://github.com/getsentry/sentry-cocoa). 

More to do later, as indicated by comments, but this all works.

Note, we require a Mac to build these targets.  I tried enabling "Pair to Mac" and building on Windows, but there are a lot of gotchas.  We can resolve that later.

### CODE REVIEW

**Comment by mattjohnsonpint** [comment]
Looks like the build server is trying to restore the workloads incorrectly.  I'll fix that.

File: src/Sentry/Platforms/iOS/SentrySdk.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            //         // Don't capture managed exceptions in the native SDK, since we already capture them in the managed SDK
+            //         o.AddIgnoredExceptionForType(JavaClass.ForName("android.runtime.JavaProxyThrowable"));
+
+        });
+
+        // Set options for the managed SDK that depend on the Cocoa SDK
+        // options.AddEventProcessor(new CocoaEventProcessor(cocoaOptions!));
+        options.CrashedLastRun = () => SentryCocoa.SentrySDK.CrashedLastRun;
+        // options.EnableScopeSync = true;
+        // options.ScopeObserver = new CocoaScopeObserver(options);
```
Comment (bruno-garcia) [inline src/Sentry/Platforms/iOS/SentrySdk.cs:130]: Save to get rid of, there's no scope sync within the cocoa SDK
```suggestion
```

**Review by bruno-garcia — COMMENTED** [review]
Great to see this landing

**Comment by mattjohnsonpint** [comment]
Passing the build now, but I had to disable the Mac Catalyst target for now.
See https://github.com/getsentry/sentry-cocoa/issues/2031

**Comment by mattjohnsonpint** [comment]
Actually, I'm going to move the Android changes to their own PR for cleanliness.

File: src/Sentry/Platforms/iOS/SentrySdk.cs (hunk 1 above)
Comment (mattjohnsonpint) [reply src/Sentry/Platforms/iOS/SentrySdk.cs:130]: Ok, I'll keep that in mind when I do the next set of these.  Leaving the comment there for now to remind me.  Thanks.

**Comment by mattjohnsonpint** [comment]
Ok, ready now.

**Review by bruno-garcia — APPROVED** [review]
Awesome stuff. Thanks Matt

**Pull request merged**